libs/blocks/merch-card/merch-card.js

```diff
@@ -102,6 +102,15 @@ function addMerchCardGridsIfMissing(section) {
   }
 }
 
+const decorateMerchCardLinkAnalytics = (el) => {
+  [...el.querySelectorAll('a')].forEach((link, index) => {
+    const heading = el.querySelector('h3');
+    const headingText = heading.textContent ? heading.textContent : '';
```

Comment:
Optional chaining might also help, in case a heading is not defined, just so you don't get the 'can't read properties of undefined' error

Reply:
Fixed